lib/k8s-e2e-tests/tests/vector-dd-agent-aggregator.rs

```diff
@@ -0,0 +1,173 @@
+"#},
+        vector_endpoint
+    );
+
+    let vector = framework
+        .vector(
+            &namespace,
+            HELM_CHART_VECTOR_AGGREGATOR,
+            VectorConfig {
+                custom_helm_values: &config_override_name(
+                    HELM_VALUES_DDOG_AGG_TOPOLOGY,
+                    &override_name,
+                ),
+                ..Default::default()
+            },
+        )
+        .await?;
+    framework
+        .wait_for_rollout(
+            &namespace,
+            &format!("statefulset/{}", override_name),
+            vec!["--timeout=60s"],
+        )
+        .await?;
+
+    let datadog_agent = framework
+        .external_chart(
+            &datadog_namespace,
+            "datadog",
+            "https://helm.datadoghq.com",
+            // VectorConfig is a generic config container
+            VectorConfig {
+                custom_helm_values: &config_override_name(
+                    datadog_chart_values,
+                    &datadog_override_name,
+                ),
+                ..Default::default()
+            },
+        )
+        .await?;
+    framework
+        .wait_for_rollout(
+            &datadog_namespace,
+            &format!("daemonset/{}", datadog_override_name),
+            vec!["--timeout=60s"],
+        )
+        .await?;
+
+    let test_namespace = framework.namespace(&pod_namespace).await?;
+    let test_pod = framework
+        .test_pod(test_pod::Config::from_pod(&make_test_pod(
+            &pod_namespace,
+            "test-pod",
+            "echo MARKER",
+            vec![],
+            // Annotation to enable log collection by the Datadog agent
+            vec![(
+                "ad.datadoghq.com/test-pod.logs",
+                "[{\"source\":\"test_source\",\"service\":\"test_service\"}]",
+            )],
+        ))?)
+        .await?;
+
+    let mut log_reader = framework.logs(&namespace, &format!("statefulset/{}", override_name))?;
+    smoke_check_first_line(&mut log_reader).await;
+
+    // Read the rest of the log lines.
+    let mut got_marker = false;
+    look_for_log_line(&mut log_reader, |val| {
+        if val["service"] != Value::Null && val["service"] != "test_service" {
+            panic!("Unexpected logs");
+        } else if val["service"] == Value::Null {
+            return FlowControlCommand::GoOn;
+        }
+
+        // Ensure we got the marker.
+        assert_eq!(val["message"], "MARKER");
+        assert_eq!(val["source_type"], "datadog_logs");
+
+        if got_marker {
+            // We've already seen one marker! This is not good, we only emitted
+            // one.
+            panic!("Marker seen more than once");
+        }
+
+        // If we did, remember it.
+        got_marker = true;
+
+        // Request to stop the flow.
+        FlowControlCommand::Terminate
+    })
+    .await?;
+
+    assert!(got_marker);
+
+    drop(test_pod);
+    drop(test_namespace);
+    drop(datadog_agent);
+    drop(vector);
```

Comment:
Any reason to drop these manually? They should just drop implicitly at the end of the scope, right?

Reply:
From what I just tested & checked I don't think there is any valid reason to keep those lines. FYI I borrowed the initial code from https://github.com/timberio/vector/blob/master/lib/k8s-e2e-tests/tests/vector.rs#L111-L113 thus there might be a reason I'm missing that *possible* explains those manual `drop`.

Reply:
I _guess_ it's to be more explicit. I don't mind keeping it in, I just wasn't sure if it was done to be explicit, or just a misunderstanding of how `drop` works.